plugin-server/src/main/ingestion-queues/session-recording-v2/sessions/session-console-log-store.ts

```diff
@@ -27,25 +35,55 @@ export class SessionConsoleLogStore {
 
     public async storeSessionConsoleLogs(logs: ConsoleLogEntry[]): Promise<void> {
         if (logs.length === 0 || !this.topic) {
-            return
+            return Promise.resolve()
         }
 
-        await this.producer.queueMessages({
-            topic: this.topic,
-            messages: logs.map((log) => ({
-                value: JSON.stringify(log),
-                key: log.log_source_id, // Using session_id as the key for partitioning
-            })),
-        })
+        if (this.pendingPromises.length >= this.promiseLimit) {
+            this.syncPromise = this.sync()
+        }
+
+        if (this.syncPromise) {
+            await this.syncPromise
+            return this.storeSessionConsoleLogs(logs)
+        }
+
+        this.pendingPromises.push(
```

Comment:
I see what you're going for. Does this have the potential for data loss? Like we could have commited a batches offsets before these promises get fully resolved?